### ISSUE TITLE
generate functions for sending messages

### ISSUE THREAD

**Issue by derberg** [asyncapi/nodejs-template#226]
After we merge https://github.com/asyncapi/nodejs-template/pull/220 the last bad user experience is 👇🏼 

```js
    client.app.send(
        { command: "off" },
        {},
        "smartylighting/streetlights/1/0/action/12/turn/on"
    );
```

we should generate

```js
function turnOn(message){
    app.send(
        message,
        {},
        "smartylighting/streetlights/1/0/action/12/turn/on"
    );
}
```

so later user can just do:

```js
turnOn({ command: "off" })
```

**Comment by derberg** [comment]
hey, thanks, but it is already in progress in https://github.com/asyncapi/nodejs-template/pull/220 PR

**Comment by PUNEET-EMM** [comment]
Okay , is their any other issue  available ?? for new comer ...

**Comment by derberg** [comment]
@PUNEET-EMM maybe have a look at https://www.asyncapi.com/community/dashboard

**Comment by kaushik-rishi** [comment]
@PUNEET-EMM #220 Has been merged, would you like to work on this issue ?

@derberg I hope we still need this.

**Comment by derberg** [comment]
yeah definitely, as for now to send a message somebody has to use client app api directly for sending, and remember topics names, instead of using a generated send functions, like we have for receiving

**Comment by ibishal** [comment]
Hey @derberg i want to solve this , but i need a little bit more clarity on the issue

**Comment by derberg** [comment]
sure, what do you need

me and @kaushik-rishi are here to help

**Comment by ibishal** [comment]
can you elaborate some more details on the issue ?
cc @derberg @kaushik-rishi

**Comment by derberg** [comment]
it's hard without specific question

you need to get this template, generate code using readme instruction and play with https://github.com/asyncapi/nodejs-template/blob/master/test/example/script.js example script.

you can see that in the example script, available generated client have helpers like `client.registerReceiveLightMeasurementMiddleware`. They are generated basing on AsyncAPI document content (examine the AsyncAPI document referred in readme, see what names are there and how them map to generated code). So it is super easy to use such a client to subscribe to topic, and provide a callback that will react to incoming messages.

Problem is with sending, to send a message you have to manually write code like
```
client.app.send(
    { command: 'off' },
    {},
    'smartylighting/streetlights/1/0/action/12/turn/on'
  );
```

as you can see, the topic address is hardcoded, and person that writes this code, need to look into docs, and make sure they use proper topic.  Better would be if client would already give you access to `turnOn` function that accepts a message payload as argument. And such function name could be generated from the operation id that in in AsyncAPI document -> https://github.com/asyncapi/nodejs-template/blob/master/test/mocks/mqtt/asyncapi-v3.yml#L111C3-L111C9

**Comment by kaushik-rishi** [comment]
@derberg fyi, i have connected with @ibishal on slack.

**Comment by nightknighto** [comment]
I'm interested in working on this issue.

It seems to be simple enough to implement:
1- update the handler.js file to also generate an additional function in case of the operation being a send operation. The generated function will be as described in the issue.
2- update the src/api/index.js to export the new function if it is a send operation.

I suggest adding `send` as prefix to differentiate between the operation ID itself and the sending function.
